### PR TITLE
[COMPARE ALL CHANGES] [encryption] Manage file key by the file to encrypt itself

### PR DESCRIPTION
**Pull request by acelyc111**

### CODE REVIEW

File: encryption/encryption.cc [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+    data_offset += actual_data_size;
+    remaining_data_size -= actual_data_size;
+  }
+
+  // TODO(yingchun): Can we remove the end partial block handling if handling a
+  //  suitable adjusted partial block at the beginning and full blocks in the
+  //  middle?
+  // Handle partial block at the end. The partial block is copied to buffer to
+  // fake a full block.
+  if (remaining_data_size > 0) {
```
Comment (WHBANG) [inline encryption/encryption.cc:304]: Can use assert to check whether `remaining_data_size` is 0.

Comment (acelyc111) [reply encryption/encryption.cc:304]: No, it's a valid case while remaining_data_size > 0.

Comment (WHBANG) [reply encryption/encryption.cc:304]: yes, but I mean use `assert` to replace `if`.

Comment (acelyc111) [reply encryption/encryption.cc:304]: Both cases (remaining_data_size > 0 and remaining_data_size == 0) are valid.

**Pull request closed without merging**